### ISSUE TITLE
Load requires parameter to exist in order to be tracked

### ISSUE THREAD

**Issue by applejuiceyy** [sveltejs/kit#5004]
### Describe the bug

Sveltekit requires the url parameter to be defined in order for it's usage to be tracked on the load function

I am guessing because it sets a getter tracking the parameter usage for all _defined_ parameters, but if a parameter is not defined, it won't track the parameter being used, and fails to re-run if it eventually gets defined without any other parameter changing

this is not inherently an issue for normal pages, but for layouts this issue is apparent

### Reproduction

root \_\_layout logging when Load runs
have a route `[first]` and another route `[first]/[second]`
travel to route `[first]/[second]`
- In this scenario, changing the `second` parameter causes a re-run on the root \_\_layout

travel to route `[first]` and back to `[first]/[second]` (the `first` parameter must not change)
- now changing the `second` parameter won't cause a re-run on the root \_\_layout, because changing to `[first]` made it not track the `second` parameter anymore

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i5-9300H CPU @ 2.40GHz    
    Memory: 2.83 GB / 11.86 GB
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.3.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (101.0.1210.47)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.42
    @sveltejs/kit: next => 1.0.0-next.330
    svelte: ^3.44.0 => 3.48.0
```

### Severity

annoyance

**Comment by babichjacob** [comment]
Please provide a reproduction adhering to the description from the bug report issue template you selected:
> A link to a repository, or a fork of https://node.new/sveltekit, that reproduces the issue. Reproductions must be [short, self-contained and correct](http://sscce.org/) and must not contain files or code that aren't relevant to the issue — please do NOT just paste a link to your project. **Explaining how to reproduce is generally not enough.** It pushes the burden of creating a reproduction project onto a small set of volunteer maintainers and isn't scalable. If no reproduction is provided, the issue will be closed.

**Comment by applejuiceyy** [comment]
I included it now
one thing I also noticed while doing this is that if you start on the top-level index page then the load function won't be re-run at all, since both parameters were undefined

another thing I noticed is that there also seems to be a bug around going back such that only the `second` parameters gets undefined doesn't trigger the load function, but that's probably out of scope for this issue

**Comment by jhubbardsf** [comment]
Yeah, I saw applejuiceyy in chat earlier with this question. If this is the intended functionality maybe the documentation should have a caveat note because it says load reruns whenever a param property changes, and it going from defined to undefined is definitely a change. I also realized that the generated types in a layout don't include optional types for params that might be there. But that's another issue. Just not sure what the intended functionality is.
